tests/integration_tests/functional/test_serial_io.py

```diff
@@ -219,3 +219,26 @@ def test_serial_block(test_microvm_with_api):
 
     # Should be significantly more than before the `cat` command.
     assert last_count - init_count > 10000
+
+
+REGISTER_FAILED_WARNING = "Failed to register serial input fd: event_manager: failed to manage epoll file descriptor: Operation not permitted (os error 1)"
+
+
+def test_no_serial_fd_error_when_daemonized(uvm_plain):
+    """
+    Tests that when running firecracker daemonized, the serial device
+    does not try to register stdin to epoll (which would fail due to stdin no
+    longer being pointed at a terminal).
+
+    Regression test for #4037.
+    """
+
+    test_microvm = uvm_plain
+    test_microvm.spawn()
+    test_microvm.basic_config(
+        vcpu_count=1,
+        mem_size_mib=512,
+    )
+    test_microvm.start()
+
```

Comment:
I would add a `test_microvm.ssh.run("true")` here to make sure the guest has finished booting.

Reply:
Ahh, I thought the .start() method will wait for it to finish booting. I'll add that, thanks! :)